deepspeed/ops/adam/cpu_adam.py

```diff
@@ -129,6 +136,9 @@ def step(self, closure=None, fp16_param_groups=None):
         # intended device for step
         device = torch.device('cpu')
 
+        if "amd" in self.cpu_vendor:
+            assert fp16_param_groups is None, "FP16 params not supported on AMD CPUs"
```

Comment:
I think we don't need to assert for this condition. The fp16_param_groups is supposed to be on GPU and it is okay on AMD.